test/trie_tests.erl

```diff
@@ -36,6 +36,113 @@ api_smoke_test_() ->
      ]
     }.
 
+root_hash_test_() ->
+    {foreach,
+     fun() ->
+	     ?debugFmt("~nCurrent working directory: ~p~n",
+		       [begin {ok, Cwd} = file:get_cwd(), Cwd end]),
+	     {ok, SupPid} =
+		 trie_sup:start_link(
+		   _KeyLength = 9,
+		   _Size = 2,
+		   _ID = ?ID,
+		   _Amount = 1000000,
+		   _Meta = 2,
+		   _HashSize = 12,
+		   _Mode = hd),
+	     ?debugFmt("~nTrie sup pid: ~p~n", [SupPid]),
+	     assert_trie_empty(0, ?ID),
+	     SupPid
+     end,
+     fun(SupPid) ->
+	     ?assert(is_process_alive(SupPid)),
+	     cleanup_alive_sup(SupPid),
+	     ?assertNot(is_process_alive(SupPid)),
+	     ok
+     end,
+     [ {"Hash of root of empty tree",
+	fun() ->
+		Root = 0,
+		RH = trie:root_hash(?ID, Root),
+		?assertMatch({RH, empty, _}, trie:get(_Key = 1, Root, ?ID)),
+		ok
+	end}
+     , {"Hash of tree root changes with tree content",
+	fun() ->
+		Root = 0,
+		RH = trie:root_hash(?ID, Root),
+		Key = 1,
+		Root2 = trie:put(Key, _V = <<1,1>>, _Meta = 0, Root, ?ID),
+		RH2 = trie:root_hash(?ID, Root2),
+		?assertMatch({RH2, _, _}, trie:get(Key, Root2, ?ID)),
+		?assertNotEqual(RH, RH2),
+		ok
+	end}
+     , {"Hash of tree root changes with leaf key",
+	fun() ->
+		Root = 0,
+		K1 = 1,
+		K2 = 2,
+		V = <<1,1>>,
+		Meta = 0,
+		Root2A = trie:put(K1, V, Meta, Root, ?ID),
+		Root2B = trie:put(K2, V, Meta, Root, ?ID),
+		?assertNotEqual(trie:root_hash(?ID, Root2A),
+				trie:root_hash(?ID, Root2B)),
+		ok
+	end}
+     , {"Hash of tree root changes with leaf value",
+	fun() ->
+		Root = 0,
+		K = 1,
+		V1 = <<1,1>>,
+		V2 = <<1,2>>,
+		Meta = 0,
+		Root2A = trie:put(K, V1, Meta, Root, ?ID),
+		Root2B = trie:put(K, V2, Meta, Root, ?ID),
+		?assertNotEqual(trie:root_hash(?ID, Root2A),
+				trie:root_hash(?ID, Root2B)),
+		ok
+	end}
+     , {"Hash of tree root does not change with leaf meta",
+	fun() ->
+		Root = 0,
+		K = 1,
+		V = <<1,1>>,
+		Meta1 = 0,
+		Meta2 = 1,
+		Root2A = trie:put(K, V, Meta1, Root, ?ID),
+		Root2B = trie:put(K, V, Meta2, Root, ?ID),
+		?assertEqual(trie:root_hash(?ID, Root2A),
```

Comment:
@zack-bitcoin in https://github.com/BumblebeeBat/MerkleTrie/blob/3214e53f2f5c4dfe4e8abb5693eab16c81b3554b/docs/todo.md I read:
> We want to keep track of some more data in each leaf. This new data should not be hashed into the merkel tree.

This test proves that that todo item can be removed, right?